protyping/auto-segmentation/create_rs_file.py

```diff
@@ -0,0 +1,202 @@
+        contour_image1.ReferencedSOPInstanceUID = (
+            "2.25.152306916480171479263212437161921183733.1"
+        )  # TODO
+        contour_image_sequence.append(contour_image1)
+
+        contour1.ContourGeometricType = "CLOSED_PLANAR"
+        contour1.ContourData = [
+            -100.5,
+            -100.5,
+            z,
+            -99.9,
+            -100.5,
+            z,
+        ]  # TODO - GET CONTOUR DATA FROM INFERENCE
+        contour1.NumberOfContourPoints = len(contour1.ContourData) // 3
+        contour_sequence.append(contour1)
+
+    roi_contour1.ReferencedROINumber = "27"
+    roi_contour_sequence.append(roi_contour1)
+
+
+def add_rt_roi_observations_sequence(ds):
+    # RT ROI Observations Sequence
+    rtroi_observations_sequence = Sequence()
+    ds.RTROIObservationsSequence = rtroi_observations_sequence
+
+    # RT ROI Observations Sequence: RT ROI Observations 1
+    rtroi_observations1 = Dataset()
+    rtroi_observations1.ObservationNumber = "27"
+    rtroi_observations1.ReferencedROINumber = "27"
+    rtroi_observations1.RTROIInterpretedType = "ORGAN"
+    rtroi_observations1.ROIInterpreter = ""
+    rtroi_observations_sequence.append(rtroi_observations1)
+
+    return ds
+
+
+dcm_paths = glob.glob(config.DATA_PATH + "/*")
+
+dcms = [pydicom.dcmread(path, force=True) for path in dcm_paths]
+
+try:
+    dcms = sorted(dcms, key=lambda dcm: dcm.SliceLocation)
+except AttributeError:
+    dcms = sorted(dcms, key=lambda dcm: dcm.SOPInstanceUID)
+
+dcm = dcms[0]
+
+ds = Dataset()
+ds.file_meta = get_file_meta(dcms)
+ds.is_implicit_VR = True
+ds.is_little_endian = True
+ds.fix_meta_info(enforce_standard=True)
+
+dt = datetime.datetime.now()
+
+ds.InstanceCreationDate = dt.strftime("%Y%m%d")
+ds.InstanceCreationTime = dt.strftime("%H%M%S.%f")
+ds.InstanceCreatorUID = "Anonymous"  # TODO
+ds.SOPInstanceUID = ds.file_meta.MediaStorageSOPInstanceUID
+ds.SOPClassUID = ds.file_meta.MediaStorageSOPClassUID
+ds.StudyDate = dcm.StudyDate
+ds.StudyTime = dcm.StudyTime
+ds.AccessionNumber = dcm.AccessionNumber
+ds.Modality = "RTSTRUCT"
+ds.Manufacturer = dcm.Manufacturer
+ds.ReferringPhysicianName = dcm.ReferringPhysicianName
+ds.InstitutionalDepartmentName = "Anonymous"  # TODO
+ds.ManufacturerModelName = dcm.ManufacturerModelName
+ds.PatientName = dcm.PatientName
+ds.PatientID = dcm.PatientID
+ds.PatientBirthDate = dcm.PatientBirthDate
+ds.PatientSex = dcm.PatientSex
+ds.StudyInstanceUID = dcm.StudyInstanceUID
+ds.StudyID = "Anonymous"  # TODO
+ds.SeriesNumber = "1"  # TODO
+ds.StructureSetLabel = "STRCTRLABEL"
+ds.StructureSetName = "STRCTRNAME"
+ds.StructureSetDate = ds.InstanceCreationDate
+ds.StructureSetTime = ds.InstanceCreationTime
+
+ds = add_referenced_frame_of_reference_sequence(ds, dcms)
+ds = add_structure_set_roi_sequence(ds)
+
+# ds = add_roi_contour_sequence(ds, dcms)
+ds = add_rt_roi_observations_sequence(ds)
+
+for x in ds:
+    print(x)
+
+ds.save_as("Test")
```

Comment:
Probably should save this file in a temp directory, or put the filename in a `.gitignore`. Good rule of thumb to never commit to version control build artefacts.

:)